### DISCUSSION TITLE
Auto Company / Auto company creation & mapping

### DISCUSSION THREAD

**Discussion by paulyili** [twentyhq/twenty#6960] [Ideas]
Ability to auto map contacts via email to existing company. 

This is a feature Hubspot has and is quite useful. 
&
This is already implemented for importing contacts via email, but i believe it should be a a global feature for manual creation & API creation of a contact. 

Basically, when adding a contact either manually or via API, Twenty should see the Email and check if the @zyx123.com exists with a company, if so to add contact to that company, if not, to create the company

**Comment by Bonapara** [comment]
I think we will address this when we improve duplicates. We'll convert it to a discussion since it won't be done in the short term. Thanks, @paulyili, for the idea!

**Comment by Marioviki** [comment]
`Can I use auto map contacts for such solution shop Viking Jewelry, Rings, Bracelets, Runes to Clothing and Weapons?`
<a href="https://vikingthunder.com/"> Viking jewelry, rings, bracelet</a>

**Comment by BOHEUS** [comment]
Related to #4954
